### ISSUE TITLE
AzureLinkAccount does not include clientSecret in API call

### ISSUE THREAD

**Issue by barclayadam** [pulumi/pulumi-newrelic#451]
### What happened?

Updating the `clientSecret` property of an `AzureLinkAccount` triggers an update and an API call, but the value is not included in the API call.

### Expected Behavior

The client secret is updated in New Relic

### Steps to reproduce

Create a resource:

````typescript
    new newrelic.cloud.AzureLinkAccount('newrelic-azure-link-account', {
        applicationId: newRelicAzureApp.applicationId,
        clientSecret: 'initial_secret',
        subscriptionId: root.subscriptionId,
        tenantId: root.tenantId,
      });
````

Update the secret:

````typescript
    new newrelic.cloud.AzureLinkAccount('newrelic-azure-link-account', {
        applicationId: newRelicAzureApp.applicationId,
        clientSecret: 'new_secret',
        subscriptionId: root.subscriptionId,
        tenantId: root.tenantId,
      });
````

See API call:

````json
{
    "query": "mutation(\n\t$accountId: Int!,\n\t$accounts: [CloudRenameAccountsInput!]!,\n) { cloudRenameAccount(\n\taccountId: $accountId,\n\taccounts: $accounts,\n) {\n\terrors {\n\t\tlinkedAccountId\n\t\tmessage\n\t\tnrAccountId\n\t\tproviderSlug\n\t\ttype\n\t}\n\tlinkedAccounts {\n\t\tauthLabel\n\t\tcreatedAt\n\t\tdisabled\n\t\texternalId\n\t\tid\n\t\tmetricCollectionMode\n\t\tname\n\t\tnrAccountId\n\t\tupdatedAt\n\t}\n} }",
    "variables":
    {
        "accountId": 123456,
        "accounts":
        [
            {
                "linkedAccountId": 987654,
                "name": "newrelic-azure-link-account-a42a668"
            }
        ]
    }
}
````

### Output of `pulumi about`

CLI
Version      3.68.0
Go Version   go1.20.4
Go Compiler  gc

Plugins
NAME             VERSION
azure            5.43.0
azure-native     1.100.0
azuread          5.26.1
cloudflare       3.6.0
docker           3.0.0
docker-buildkit  0.1.17
kubernetes       3.23.1
newrelic         5.10.0
nodejs           unknown
random           4.0.0
tls              4.0.0

Host
OS       Microsoft Windows 11 Pro
Version  10.0.22000 Build 22000
Arch     x86_64

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by guineveresaenger** [comment]
Hi @barclayadam - thank you for filing this issue.

Could you verify if this behavior persists on the latest version of newrelic (`v5.12.0`)? We've updated a few things concerning secrets handling since.

Thank you so much! 🙏

**Comment by barclayadam** [comment]
Hi @guineveresaenger 

I've confirmed that the same behaviour persists on v5.12.0

Also to add I'm using Automation API and TypeScript

**Comment by guineveresaenger** [comment]
Thank you for double checking!

Is it possible for you to create a minimal repro of this? I suspect that since an update is triggered by Pulumi, as expected, that this is an upstream API limitation.

Due to the [extra configuration requirements](https://www.pulumi.com/registry/packages/newrelic/api-docs/cloud/azurelinkaccount/) for this integration, I suspect that this is unimplemented due to API limitations when reading/sending sensitive information.

**Comment by guineveresaenger** [comment]
Update: This is [not implemented upstream](https://github.com/newrelic/terraform-provider-newrelic/blob/main/newrelic/resource_newrelic_cloud_azure_link_account.go#L147) and I have filed an issue.

**Comment by mikhailshilkov** [comment]
Fixed upstream, the fix is available in [5.15.1](https://github.com/pulumi/pulumi-newrelic/releases/tag/v5.15.1)
